src/Microsoft.Data.Analysis/DataFrame.IO.cs

```diff
@@ -306,5 +309,123 @@ private static DataFrameColumn CreateColumn(Type kind, string[] columnNames, int
+        /// <param name="path">CSV file path</param>
+        /// <param name="separator">column separator</param>
+        /// <param name="header">has a header or not</param>
+        /// <param name="encoding">The character encoding. Defaults to UTF8 if not specified</param>
+        /// <param name="cultureInfo">culture info for handeling decimal values</param>
+        public static void WriteCsv(DataFrame dataFrame, string path,
+                                   char separator = ',', bool header = true,
+                                   Encoding encoding = null, CultureInfo cultureInfo = null)
+        {
+            using (FileStream csvStream = new FileStream(path, FileMode.Create))
+            {
+                WriteCsv(dataFrame: dataFrame, csvStream: csvStream,
+                           separator: separator, header: header,
+                           encoding: encoding, cultureInfo: cultureInfo);
+            }
+        }
+
+        /// <summary>
+        /// Writes a DataFrame into a CSV.
+        /// </summary>
+        /// <param name="dataFrame"><see cref="DataFrame"/></param>
+        /// <param name="csvStream">stream of CSV data to be write out</param>
+        /// <param name="separator">column separator</param>
+        /// <param name="header">has a header or not</param>
+        /// <param name="encoding">the character encoding. Defaults to UTF8 if not specified</param>
+        /// <param name="cultureInfo">culture info for handeling decimal values</param>
+        public static void WriteCsv(DataFrame dataFrame, Stream csvStream,
+                           char separator = ',', bool header = true,
+                           Encoding encoding = null, CultureInfo cultureInfo = null)
+        {
+            if (cultureInfo is null)
+            {
+                cultureInfo = CultureInfo.CurrentCulture;
+            }
+
+            if (cultureInfo.NumberFormat.NumberDecimalSeparator.Equals(separator.ToString()))
+            {
+                throw new ArgumentException("Decimal separator cannot match the column separator");
+            }
+
+            if (encoding is null)
+            {
+                encoding = Encoding.ASCII;
+            }
+
+            using (StreamWriter csvFile = new StreamWriter(csvStream, encoding, bufferSize: DefaultStreamReaderBufferSize, leaveOpen: true))
+            {
+                if (dataFrame != null)
+                {
+                    var columnNames = dataFrame.Columns.GetColumnNames();
+
+                    if (header)
+                    {
+                        var headerColumns = string.Join(separator.ToString(), columnNames);
+                        csvFile.WriteLine(headerColumns);
+                    }
+
+                    var record = new StringBuilder();
+
+                    foreach (var row in dataFrame.Rows)
+                    {
+                        bool firstRow = true;
```

Comment:
```suggestion
                        bool firstColumn = true;
```

or 

```suggestion
                        bool firstColumnInRow = true;
```

Reply:
Agree with the rename. This is what was confusing me the last time I reviewed this :).

Reply:
> Agree with the rename. This is what was confusing me the last time I reviewed this :).

It makes sense!